atc/exec/task_config_source_test.go

```diff
@@ -212,19 +222,19 @@ run: {path: a/file}
 
 			Context("when the file task is not found", func() {
 				BeforeEach(func() {
-					fakeArtifactSource.StreamFileReturns(nil, baggageclaim.ErrFileNotFound)
+					fakeWorkerClient.StreamFileFromArtifactReturns(nil, baggageclaim.ErrFileNotFound)
 				})
 
 				It("returns the error", func() {
 					Expect(fetchErr).To(HaveOccurred())
-					Expect(fetchErr.Error()).To(Equal("task config 'some/build.yml' not found"))
+					Expect(fetchErr.Error()).To(Equal(fmt.Sprintf("task config '%s/build.yml' not found", artifactName)))
 				})
 			})
 		})
 
 		Context("when the file's artifact source cannot be found in the repository", func() {
 			It("returns an UnknownArtifactSourceError", func() {
-				Expect(fetchErr).To(Equal(UnknownArtifactSourceError{SourceName: "some", ConfigPath: "some/build.yml"}))
+				Expect(fetchErr).To(Equal(UnknownArtifactSourceError{SourceName: build.ArtifactName(artifactName), ConfigPath: artifactName + "/build.yml"}))
 			})
 		})
 	})
```

Comment:
It's not letting me put an inline comment further down for some reason, but I think that with the change from pointers to values nested in the `TaskConfigSource`, it's worth adding an extra assertion to the unit test literally at the bottom of this entire file to confirm that `fetchedConfig.ImageResource.Params` also changes to take the interpolated values.